torchao/testing/model_architectures.py

```diff
@@ -11,14 +11,74 @@
 import torch.nn.functional as F
 
 
-# TODO: Refactor torchao and tests to use these models
-class ToyLinearModel(torch.nn.Module):
-    def __init__(self, k=64, n=32, dtype=torch.bfloat16):
+class ToySingleLinearModel(torch.nn.Module):
+    """Single linear for m * k * n problem size"""
+
+    def __init__(
+        self, m=64, n=32, k=64, has_bias=False, dtype=torch.float, device="cuda"
+    ):
+        super().__init__()
+        self.m = m
+        self.dtype = dtype
+        self.device = device
+        self.linear = torch.nn.Linear(k, n, bias=has_bias).to(
+            dtype=self.dtype, device=self.device
+        )
+
+    def example_inputs(self):
+        return (
+            torch.randn(
+                self.m, self.linear.in_features, dtype=self.dtype, device=self.device
+            ),
+        )
+
+    def forward(self, x):
+        x = self.linear(x)
+        return x
+
+
+class ToyTwoLinearModel(torch.nn.Module):
+    def __init__(
+        self,
+        input_dim,
+        hidden_dim,
+        output_dim,
+        has_bias=False,
+        dtype=torch.bfloat16,
+        device="cpu",
+    ):
         super().__init__()
-        self.linear1 = torch.nn.Linear(k, n, bias=False).to(dtype)
+        self.dtype = dtype
+        self.device = device
+        self.linear1 = torch.nn.Linear(
+            input_dim, hidden_dim, bias=has_bias, dtype=dtype, device=device
+        )
+        self.linear2 = torch.nn.Linear(
+            hidden_dim, output_dim, bias=has_bias, dtype=dtype, device=device
+        )
+
+    def example_inputs(
+        self, batch_size=1, sequence_length=None, dtype=None, device=None
+    ):
+        if dtype is None:
+            dtype = self.dtype
+        if device is None:
+            device = self.device
+
+        if sequence_length is not None:
```

Comment:
is this used? this can be done in callsite as well right?

Reply:
can we move sequence_length to the call site? seems like it's the loop can happen outside